File: (June)30_days_Challenge/reverseString.py
# Write a function that reverses a string. The input string is given as an array of characters char[].
# Do not allocate extra space for another array, you must do this by modifying the input array in-place with O(1) extra memory.
# You may assume all the characters consist of printable ascii characters.

from typing import List
class Solution:
    def reverseString(self, s: List[str]) -> None:
        size = len(s)
        if size == 0 or size == 1: return
        i = 0
        j = size-1
        while(i < j):
            s[i], s[j] = s[j], s[i]
            i+=1
            j-=1

File: (June)30_days_Challenge/reverseString.cpp
// Write a function that reverses a string. The input string is given as an array of characters char[].
// Do not allocate extra space for another array, you must do this by modifying the input array in-place with O(1) extra memory.
// You may assume all the characters consist of printable ascii characters.

#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    void reverseString(vector<char>& s) {
        int size = s.size();
        if(size == 0 || size == 1) return;
        int i = 0;
        int j = size-1;
        while(i < j)
        {
            char temp = s[i];
            s[i] = s[j];
            s[j] = temp;
            i++;
            j--;
        }
    }
};